### PR TITLE
Metabase, lambda and refactoring

### PR DESCRIPTION
**Pull request by jarri-abidi**

### CODE REVIEW

File: cmd/mock/main.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"net/http"
+	"os"
+)
+
+func main() {
+	addr := flag.String("addr", ":56000", "listen addr")
+	flag.Parse()
+
+	if err := http.ListenAndServe(*addr, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		f, err := os.Open("karma/trips_sample.json")
```
Comment (markhaur) [inline cmd/mock/main.go:16]: don't you think we have forgot to `defer file.Close()`???

File: karma/trips.go [hunk 2]
```diff
@@ -42,16 +42,17 @@ type Trip struct {
 	TripDistanceNight int32   `json:"trip_distance_night"`
 }
 
-func FetchTrips(ctx context.Context, url string, req FetchTripsRequest) (*FetchTripsResponse, error) {
+func (c *Client) FetchTrips(ctx context.Context, req FetchTripsRequest) (*FetchTripsResponse, error) {
```
Comment (markhaur) [inline karma/trips.go]: I don't think there is any need for using `Client` pointer since we are reading `url` field.

**Review by markhaur — COMMENTED** [review]
Kindly see the comments added!

File: cmd/mock/main.go (hunk 1 above)
Comment (jarri-abidi) [reply cmd/mock/main.go:16]: add a code suggestion and we can commit it

File: karma/trips.go (hunk 2 above)
Comment (jarri-abidi) [reply karma/trips.go]: add a code suggestion with (c Client)

Comment (markhaur) [reply karma/trips.go]:
```suggestion
func (c Client) FetchTrips(ctx context.Context, req FetchTripsRequest) (*FetchTripsResponse, error) {
```

File: cmd/mock/main.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+		if err != nil {
+			log.Printf("could not read file: %v", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
```
Comment (markhaur) [inline cmd/mock/main.go]:
```suggestion
defer f.Close()
```

**Review by markhaur — APPROVED** [review]

**Pull request merged**